trunk/SUAVE/Components/Energy/Networks/Battery_Propeller.py

```diff
@@ -90,151 +98,263 @@ def evaluate_thrust(self,state):
+                conditions.propulsion.disc_loading[:,ii]               = (F_mag[:,0])/(np.pi*(R**2)) # N/m^2                  
+                conditions.propulsion.power_loading[:,ii]              = (F_mag[:,0])/(P[:,0])      # N/W      
+                conditions.propulsion.propeller_efficiency[:,ii]       = etap[:,0]      
+                    
+                conditions.noise.sources.propellers[prop.tag]          = outputs
+    
+            # Run the avionics
+            avionics.power()
+    
+            # Run the payload
+            payload.power()
+            
+            # link
+            esc.inputs.currentout = total_motor_current
     
+            # Run the esc
+            esc.currentin(conditions)  
+            
+            # Calculate avionics and payload power
+            avionics_payload_power = avionics.outputs.power + payload.outputs.power
+        
+            # Calculate avionics and payload current
+            avionics_payload_current = avionics_payload_power/self.voltage 
+        
+            # link
+            battery.inputs.current  = esc.outputs.currentin + avionics_payload_current
+            battery.inputs.power_in = -(esc.outputs.voltageout *esc.outputs.currentin + avionics_payload_power)
+            battery.energy_cycle_model(numerics,discharge_flag)         
+             
+        # --------------------------------------------------------------------------------
+        # Run Charge Model 
+        # --------------------------------------------------------------------------------               
+        else:  
+            # link 
+            battery.inputs.current  = -battery.cell.charging_current*n_parallel * np.ones_like(volts)
+            battery.inputs.voltage  =  battery.cell.charging_voltage*n_series * np.ones_like(volts)
+            battery.inputs.power_in =  -battery.inputs.current * battery.inputs.voltage             
+            battery.energy_cycle_model(numerics,discharge_flag)        
+            
+            avionics_payload_current = np.zeros((len(volts),1)) 
+            total_thrust             = np.zeros((len(volts),3)) 
+            P                        = battery.inputs.power_in
+            
         # Pack the conditions for outputs
-        conditions.propulsion.battery_current              = esc.outputs.currentin
-        conditions.propulsion.battery_draw                 = battery.inputs.power_in 
-        conditions.propulsion.battery_energy               = battery.current_energy
-        conditions.propulsion.battery_voltage_open_circuit = battery.voltage_open_circuit
-        conditions.propulsion.battery_voltage_under_load   = battery.voltage_under_load
-        conditions.propulsion.state_of_charge              = battery.state_of_charge
-        conditions.propulsion.battery_specfic_power        = -battery.inputs.power_in/battery.mass_properties.mass #Wh/kg
-    
-        # Create the outputs
+        battery_draw = battery.inputs.power_in    
```

Comment:
how about a battery "packing" function that we can reuse with all the electric networks?

Reply:
packaged